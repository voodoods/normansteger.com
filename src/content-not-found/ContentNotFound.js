import React, { Component } from 'react'
import StyledContentNotFound from './ContentNotFound.styled'

import brokenRobot from './robot_broken.svg'

class ContentNotFound extends Component {
  dropReport() {
    const email = 'me@normansteger.com'
    const subject = 'Missing content on normansteger.com'
    const emailBody = `Hi there, \n I just encountered a part of missing content on your homepage at he URL: ${
      window.location.href
    } \n Please have a look at it as soon as you can. Thank you!`
    document.location = 'mailto:' + email + '?subject=' + subject + '&body=' + emailBody
  }

  render() {
    return (
      <StyledContentNotFound className="ContentNotFound">
        <img src={brokenRobot} alt="Content not Found" />
        <h3>Uh oh... this isn&lsquo;t supposed to happen :(</h3>

        <p>
          If you read this, please take a moment to&nbsp;
          <span onClick={() => this.dropReport()}>drop me a message</span>. I&lsquo;d be most greatful and will fix the
          issue as soon as I get back to the keyboard.
        </p>
      </StyledContentNotFound>
    )
  }
}

export default ContentNotFound
