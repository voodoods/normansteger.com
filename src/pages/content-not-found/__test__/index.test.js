import React from 'react'
import { shallow } from 'enzyme'
import ContentNotFound from '../'

describe('<ContentNotFound /> component index', () => {
  it('expects component to match snapshot', () => {
    const wrapper = shallow(<ContentNotFound />).dive()
    expect(wrapper).toMatchSnapshot()
  })
})
