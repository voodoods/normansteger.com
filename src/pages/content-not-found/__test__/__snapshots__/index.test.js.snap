// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentNotFound /> component index expects component to match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContentNotFound
    dropReport={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <img
          alt="Content not Found"
          src="robot_broken.svg"
        />,
        <h3>
          Uh oh... this isn‘t supposed to happen :(
        </h3>,
        <p>
          If you read this, please take a moment to 
          <span
            onClick={[Function]}
          >
            drop me a message
          </span>
          . I‘d be most greatful and will fix the issue as soon as I get back to the keyboard.
        </p>,
      ],
      "className": "ContentNotFound",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Content not Found",
          "src": "robot_broken.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Uh oh... this isn‘t supposed to happen :(",
        },
        "ref": null,
        "rendered": "Uh oh... this isn‘t supposed to happen :(",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "If you read this, please take a moment to ",
            <span
              onClick={[Function]}
            >
              drop me a message
            </span>,
            ". I‘d be most greatful and will fix the issue as soon as I get back to the keyboard.",
          ],
        },
        "ref": null,
        "rendered": Array [
          "If you read this, please take a moment to ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "drop me a message",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "drop me a message",
            "type": "span",
          },
          ". I‘d be most greatful and will fix the issue as soon as I get back to the keyboard.",
        ],
        "type": "p",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  width: 50%;

  img {
    margin-top: 3em;
    width: 50%;
  }

  h3 {
    margin: 0;
  }

  span {
    color: blue;
    font-weight: bold;
    text-decoration: underline;
    cursor: pointer;
    &:hover {
      opacity: 0.6;
    }
  }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <img
            alt="Content not Found"
            src="robot_broken.svg"
          />,
          <h3>
            Uh oh... this isn‘t supposed to happen :(
          </h3>,
          <p>
            If you read this, please take a moment to 
            <span
              onClick={[Function]}
            >
              drop me a message
            </span>
            . I‘d be most greatful and will fix the issue as soon as I get back to the keyboard.
          </p>,
        ],
        "className": "ContentNotFound",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Content not Found",
            "src": "robot_broken.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Uh oh... this isn‘t supposed to happen :(",
          },
          "ref": null,
          "rendered": "Uh oh... this isn‘t supposed to happen :(",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "If you read this, please take a moment to ",
              <span
                onClick={[Function]}
              >
                drop me a message
              </span>,
              ". I‘d be most greatful and will fix the issue as soon as I get back to the keyboard.",
            ],
          },
          "ref": null,
          "rendered": Array [
            "If you read this, please take a moment to ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "drop me a message",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "drop me a message",
              "type": "span",
            },
            ". I‘d be most greatful and will fix the issue as soon as I get back to the keyboard.",
          ],
          "type": "p",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  width: 50%;

  img {
    margin-top: 3em;
    width: 50%;
  }

  h3 {
    margin: 0;
  }

  span {
    color: blue;
    font-weight: bold;
    text-decoration: underline;
    cursor: pointer;
    &:hover {
      opacity: 0.6;
    }
  }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
